src/main/src/main/java/org/geoserver/catalog/LayerGroupHelper.java

```diff
@@ -241,7 +242,18 @@ private static void allStylesForRendering(
                     PublishedInfo p = group.getLayers().get(i);
                     StyleInfo s = group.getStyles().get(i);
                     if (p instanceof LayerInfo) {
-                        styles.add(group.getStyles().get(i));
```

Comment:
Why doesn't the ``group.getStyles()`` already contains the correct style?  When a layer is added the layer group a style is selected right? at that moment it would make  sense to select the right style for remote layers too.

Reply:
because the style group does not know about the layer default